core/src/main/java/org/apache/druid/java/util/common/HumanReadableBytes.java

```diff
@@ -159,10 +159,18 @@ private static long parseInner(String rawNumber)
         throw new IAE("Invalid format of number: %s", rawNumber);
       }
 
+      unit = number.charAt(lastDigitIndex--);
+      isBinaryByte = true;
+    } else if (unit == 'i') {
+      //unit ends with 'i' must be format of Ki/Mi/Gi/Ti/Pi, so at least 2 extra characters are required
+      if (lastDigitIndex < 1) {
+        throw new IAE("Invalid format of number: %s", rawNumber);
```

Comment:
Exception messages above are not instructive, so I think it's better to improve it as something like
```suggestion
        throw new IAE("Invalid format of number [%s]. The unit should be one of Pi/Ti/Gi/Mi/Ki", rawNumber);
```